staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -633,9 +633,13 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 	}
 
 	out := e.NewFunc()
+
+	// only ignore a not found error if this type allows creating on update, or we're forcing allowing create (like for server-side-apply)
+	ignoreNotFound := e.UpdateStrategy.AllowCreateOnUpdate() || forceAllowCreate
+
 	// deleteObj is only used in case a deletion is carried out
 	var deleteObj runtime.Object
-	err = e.Storage.GuaranteedUpdate(ctx, key, out, true, storagePreconditions, func(existing runtime.Object, res storage.ResponseMeta) (runtime.Object, *uint64, error) {
```

Comment:
passing `true` here unconditionally means that when we *can't* find an existing object in etcd, the missing uid or resourceVersion will trip a precondition error, rather than getting into the callback below that returns a NotFound error